Objects/codeobject.c

```diff
@@ -1420,8 +1398,21 @@ code_richcompare(PyObject *self, PyObject *other, int op)
     if (!eq) goto unequal;
     eq = co->co_firstlineno == cp->co_firstlineno;
     if (!eq) goto unequal;
-    eq = PyObject_RichCompareBool(co->co_code, cp->co_code, Py_EQ);
-    if (eq <= 0) goto unequal;
+    PyObject *co_code = _PyCode_GetCode(co);
```

Comment:
No need to create both code objects here. Iterating over both bytecode arrays, comparing `_PyOpcode_Deopt[op]` and skipping caches would be considerably more efficient.

Added to https://github.com/faster-cpython/ideas/issues/310